src/org/labkey/test/components/ui/grids/DetailTable.java

```diff
@@ -120,17 +147,33 @@ public void clickField(String fieldCaption)
     {
         static final Locator.XPathLocator detailTable = Locator.tagWithClass("table", "detail-component--table__fixed");
 
-        static Locator fieldValue(String caption)
-        {
-            return Locator.tagWithAttribute("td", "data-caption", caption);
-        }
-
         static final Locator loadingGrid = Locator.css("tbody tr.grid-loading");
         static final Locator emptyGrid = Locator.css("tbody tr.grid-empty");
         static final Locator spinner = Locator.css("span i.fa-spinner");
 
     }
 
+    @Override
+    protected ElementCache newElementCache()
+    {
+        return new ElementCache();
+    }
+
+    protected class ElementCache extends Component<?>.ElementCache
+    {
+        public final WebElement dataCaptionField(String caption)
+        {
+            return Locator.tagWithAttribute("td", "data-caption", caption).findWhenNeeded(this);
+        }
+
+        // Some tables will show a value in a td with no attributes, use the td that has the text (label) to find the value.
+        public final WebElement siblingField(String caption)
+        {
+            return Locator.tagContainingText("td", caption).followingSibling("td").findWhenNeeded(this);
+        }
```

Comment:
Consider unioning these Locators if there's no need to distinguish between the two.
```suggestion
        public final WebElement dataCaptionField(String caption)
        {
            return Locator.XPathLocator.union(
                Locator.tagWithAttribute("td", "data-caption", caption),
                Locator.tagContainingText("td", caption).followingSibling("td")) // Some tables will show a value in a td with no attributes, use the td that has the text (label) to find the value.
            .findElement(this);
        }
```
Also, is there some distinguishing CSS class or attribute that we could add to the sibling Locator? (e.g. `Locator.tagContainingText("td", caption).followingSibling("td").withAttribute("data-caption")`)

Reply:
Unfortunately there is not a consistent attribute in the sibling locator. It can range form what you have mentioned (data-caption) to "detail-display" to a td with no attributes.